File: Data Structures & Algorithms/longest-consecutive-sequence/submission-18.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        var set = new HashSet<int>(nums);
        var res = 0;
        for (int i = 0; i < nums.Length; i++) {
            if (set.Contains(nums[i] - 1)) {
                continue;
            }
            var length = 1;
            while (set.Contains(nums[i] + length)) {
                length++;
            }
            res = Math.Max(res, length);
        }
        return res;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-18.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)
        l, r = 0, n - 1
        res = float('-inf')
        while l < r:
            m = l + ((r - l) // 2)
            mid = nums[m]
            if nums[l] > nums[r]:
                if nums[l] <= nums[m]:
                    l = m + 1
                else:
                    r = m
            else:
                r = m
        return nums[l]
